sdk/src/Services/S3/Custom/Transfer/Internal/_async/MultipartUploadCommand.async.cs

```diff
@@ -219,14 +220,14 @@ private void AbortMultipartUpload(string uploadId)
                         int partNumber = 1;
                         int readBytesCount, readAheadBytesCount = 0;
 
-                        readBytesCount = await stream.ReadAsync(readBuffer, 0, readBuffer.Length).ConfigureAwait(false);
+                        readBytesCount = await stream.ReadAsync(readBuffer, 0, readBuffer.Length, cancellationToken).ConfigureAwait(false);
 
                         do
                         {
                             await nextUploadBuffer.WriteAsync(readBuffer, 0, readBytesCount).ConfigureAwait(false);
```

Comment:
@mmaderic You are missing passing `CancellationToken` in `nextUploadBuffer.WriteAsync()`.